File: packages/pdfx/android/src/main/kotlin/io/scer/pdfx/resources/PageRepository.kt
package io.scer.pdfx.resources

import android.annotation.TargetApi
import android.graphics.pdf.PdfRenderer
import android.os.Build
import io.scer.pdfx.document.Page
import io.scer.pdfx.utils.randomID

@TargetApi(Build.VERSION_CODES.LOLLIPOP)
class PageRepository : Repository<Page>() {
    /**
     * Register page in repository
     * @returns page id
     */
    fun register(documentId: String, pageRenderer: PdfRenderer.Page): Page {
        val id = randomID
        val page = Page(id, documentId, pageRenderer)
        set(id, page)
        return page
    }

    public override fun close(id: String) {
        get(id).close()
        super.close(id)
    }
}

File: packages/pdfx/lib/src/renderer/web/resources/page_repository.dart
import 'package:pdfx/src/renderer/web/document/page.dart';
import 'package:pdfx/src/renderer/web/pdfjs.dart';
import 'package:pdfx/src/renderer/web/resources/repository.dart';

class PageRepository<T> extends Repository<Page> {
  /// Register document in repository
  Page register(String? documentId, PdfjsPage renderer) {
    final page = Page(
      id: uuid.v1(),
      documentId: documentId,
      renderer: renderer,
    );
    set(page.id, page);
    return page;
  }

  @override
  void close(String? id) {
    get(id)!.close();
    super.close(id);
  }
}
